pytealext/evaluator/tools.py

```diff
@@ -1,11 +1,60 @@
+import re
+from typing import IO
+
 from pyteal import MAX_PROGRAM_VERSION, Expr, Mode, compileTeal
 
-from .evaluator import eval_teal
+from .evaluator import EvalContext, eval_teal
+
+
+def substitute_template_values(teal: str, substitutions: dict[str, str]) -> str:
+    """Replaces all template variables (those starting with `TMPL_`)"""
+
+    def repl(m: re.Match) -> str:
+        return substitutions.get(m[0], m[0])
+
+    return re.sub(r"\bTMPL_\w+\b", repl, teal)
 
 
 def compile_and_run(
-    ast: Expr, mode=Mode.Application, version=MAX_PROGRAM_VERSION, *, context=None, debug=None
+    ast: Expr,
+    mode: Mode = Mode.Application,
+    version: int = MAX_PROGRAM_VERSION,
+    *,
+    context: EvalContext = None,
+    debug: IO = None,
+    tmpl_subs: dict[str, str] = None
 ) -> tuple[list[bytes or int], list[bytes or int]]:
```

Comment:
why asterix in this place

Reply:
I think it would be hard to maintain this long list of positional arguments, also, the order wouldn't make sense. If you're asking why specifically this position, then it doesn't really make sense

The only good idea here is to keep mode positional, so it is the same as in compileTeal

Reply:
WDYT about moving the asterisk after mode?